### PR TITLE
chore(swap): handle swap balance errors

### PR DESCRIPTION
**Pull request by banklesss**
I refactored a bit to disable swap button in case of any balance error.

Related to [YOMO-819](https://emurgo.atlassian.net/browse/YOMO-819)

[YOMO-819]: https://emurgo.atlassian.net/browse/YOMO-819?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx [hunk 1]
```diff
@@ -239,6 +251,77 @@ export const CreateOrder = () => {
...
+
+  const hasBalance = !Quantities.isGreaterThan(
+    Quantities.sum([
+      tokenId === wallet.primaryTokenInfo.id ? orderData.amounts.sell.quantity : Quantities.zero,
+      orderData.selectedPoolCalculation?.cost.batcherFee.quantity ?? Quantities.zero,
+      orderData.selectedPoolCalculation?.cost.frontendFeeInfo.fee.quantity ?? Quantities.zero,
+      orderData.selectedPoolCalculation?.pool.deposit.quantity ?? Quantities.zero,
+    ]),
+    balance,
+  )
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]: Maybe here we need to separate primary token amount and other token amount right?

two types of balance: primary token balance + other types

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx [hunk 2]
```diff
@@ -239,6 +251,84 @@ export const CreateOrder = () => {
...
+
+  const hasSecondaryTokenBalance = !Quantities.isGreaterThan(
+    tokenId !== wallet.primaryTokenInfo.id ? orderData.amounts.sell.quantity : Quantities.zero,
+    balance,
+  )
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]: let me know what do you think about this

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx (hunk 1 above)
Comment (jorbuedo) [reply apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]: @banklesss if you need all fees summed up, add them to makeOrderCalculations in another `cost` field, so you don't have to sum them in the app's front

Comment (jorbuedo) [reply apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]: @stackchain we need to clarify if the frontendfee should be added or not
We said let's calculate it so we know how much it is, but we don't apply it
If we're not applying it it shouldn't be summed up here

Comment (banklesss) [reply apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]: `ptTotalFee` added to `orderData.selectedPoolCalculation?.cost`

Comment (stackchain) [reply apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]: the same way we have the other feeds, let's have the ptTotalFeeWithFeF and without

Comment (stackchain) [reply apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]: and the client decides

File: apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx [hunk 3]
```diff
@@ -70,7 +73,11 @@ export const CreateOrder = () => {
       setShowLimitPriceWarning(false)
     },
     onError: (error) => {
-      console.log(error)
+      if (error instanceof NotEnoughMoneyToSendError) {
+        setSellBackendError(strings.notEnoughBalance)
+      } else {
+        console.error(error)
+      }
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx:80]:
```suggestion
      }
```

Comment (stackchain) [inline apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx]: maybe fallback to an `unkonwon error` if the api/network fails?

**Review by michalszorad — APPROVED** [review]

**Review by stackchain — APPROVED** [review]

**Pull request merged**